go/cmd/dolt/commands/rebase.go

```diff
@@ -0,0 +1,369 @@
+			if err != nil {
+				return HandleVErrAndExitCode(errhand.VerboseErrorFromError(err), usage)
+			}
+
+			rows, err := GetRowsForSql(queryist, sqlCtx, "CALL DOLT_REBASE('--continue');")
+			if err != nil {
+				return HandleVErrAndExitCode(errhand.VerboseErrorFromError(err), usage)
+			}
+			status, err := getInt64ColAsInt64(rows[0][0])
+			if err != nil {
+				return HandleVErrAndExitCode(errhand.VerboseErrorFromError(err), usage)
+			}
+			if status == 1 {
+				return HandleVErrAndExitCode(errhand.VerboseErrorFromError(errors.New("error: "+rows[0][1].(string))), usage)
+			}
+
+			cli.Println("Successfully rebased and updated refs/heads/" + branchName + ".")
+		}
+	}
+
+	return HandleVErrAndExitCode(nil, usage)
+}
+
+// constructInterpolatedDoltRebaseQuery generates the sql query necessary to call the DOLT_REBASE() function.
+// Also interpolates this query to prevent sql injection.
+func constructInterpolatedDoltRebaseQuery(apr *argparser.ArgParseResults, rebaseBranchName string) (string, error) {
+	var params []interface{}
+	var args []string
+	var checkoutQuery string
+
+	if apr.NArg() == 1 {
+		params = append(params, apr.Arg(0))
+		args = append(args, "?")
+	}
+	if apr.Contains(cli.InteractiveFlag) {
+		args = append(args, "'--interactive'")
+	}
+	if apr.Contains(cli.ContinueFlag) {
+		args = append(args, "'--continue'")
+	}
+	if apr.Contains(cli.AbortParam) {
+		args = append(args, "'--abort'")
+	}
+
+	/*if apr.ContainsAny(cli.ContinueFlag, cli.AbortParam) {
+		checkoutQuery = fmt.Sprintf("CALL DOLT_CHECKOUT('dolt_rebase_%s'); ", rebaseBranchName)
+	}*/
+
+	query := fmt.Sprintf("%sCALL DOLT_REBASE(%s);", checkoutQuery, strings.Join(args, ", "))
+	interpolatedQuery, err := dbr.InterpolateForDialect(query, params, dialect.MySQL)
+	if err != nil {
+		return "", err
+	}
+
+	return interpolatedQuery, nil
+}
+
+// getRebasePlan opens an editor for users to edit the rebase plan and returns the parsed rebase plan from the editor.
+func getRebasePlan(cliCtx cli.CliContext, sqlCtx *sql.Context, queryist cli.Queryist, rebaseBranch, currentBranch string) (*rebase.RebasePlan, error) {
+	if cli.ExecuteWithStdioRestored == nil {
+		return nil, nil
+	}
+
+	if !checkIsTerminal() {
+		return nil, nil
+	}
+
+	initialRebaseMsg, err := buildInitialRebaseMsg(sqlCtx, queryist, rebaseBranch, currentBranch)
+	if err != nil {
+		return nil, err
+	}
+
+	backupEd := "vim"
+	// try getting default editor on the user system
+	if ed, edSet := os.LookupEnv(dconfig.EnvEditor); edSet {
+		backupEd = ed
+	}
+	// try getting Dolt config core.editor
+	editorStr := cliCtx.Config().GetStringOrDefault(config.DoltEditor, backupEd)
+
+	var rebaseMsg string
+	cli.ExecuteWithStdioRestored(func() {
+		rebaseMsg, err = editor.OpenTempEditor(editorStr, initialRebaseMsg)
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	rebasePlan, err := parseRebaseMessage(rebaseMsg)
+	if err != nil {
+		return nil, err
+	}
+
+	return rebasePlan, nil
```

Comment:
(minor) Similar comment here. This could be simplified to just `return parseRebaseMessage(rebaseMsg)`